test/denormalized_has_many_test.rb

```diff
@@ -94,24 +94,29 @@ def test_saving_associated_records_should_expire_itself_and_the_parents_cache
     child.save!
   end
 
-  def test_unsupported_through_assocation
-    assert_raises IdentityCache::UnsupportedAssociationError, "caching through associations isn't supported" do
-      Item.has_many :deeply_through_associated_records, :through => :associated_records, foreign_key: 'associated_record_id', inverse_of: :item, :class_name => 'DeeplyAssociatedRecord'
-      Item.cache_has_many :deeply_through_associated_records, :embed => true
+  class CheckAssociationTest < IdentityCache::TestCase
```

Comment:
`@record.save` in the setup would cause the cache key to be memoized, so I wanted to use an empty setup at least for test_unsupported_joins_in_assocation_scope.